katcp/inspecting_client.py

```diff
@@ -519,13 +519,38 @@ def handle_sensor_value(self):
 
     @tornado.gen.coroutine
     def inspect(self):
-        """Inspect device requests and sensors, update model"""
+        """Inspect device requests and sensors, update model
+
+        Returns
+        -------
+
+        Tornado future that resolves with:
+
+        model_changes : Nested AttrDict
+            Contanins sets of added/removed request/sensor names
+
+            Example structure:
+
+            {'requests': {
+                'added': set(['req1', 'req2']),
+                'removed': set(['req10', 'req20'])}
+             'sensors': {
+                'added': set(['sens1', 'sens2']),
+                'removed': set(['sens10', 'sens20'])}
+            }
+
+            If there are no changes keys may be omitted. If an item is in both
+            the 'added' and 'removed' sets that means that it changed.
```

Comment:
This comment sounds like you will get an empty dict if there were no changes at all.  
Suggest something like:

> - If an item is in both the 'added' and 'removed' sets that means that it changed.
> - If some, but not all types of changes occurred, keys may be omitted for 
> items that did not change.
> - If there are neither request nor sensor changes, resolves with None